packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -51,6 +53,21 @@ class FlutterManifest {
 
   String get appName => _descriptor['name'] ?? '';
 
+  /// The version String from the `pubspec.yaml` file. Can be null.
+  String get appVersion {
+    final String version = _descriptor['version']?.toString();
+    return version?.contains(_versionPattern) ?? false ? version : null;
+  }
+
+  /// The build version name from the `pubspec.yaml` file. Can be null.
+  String get buildName => appVersion?.split('+')?.elementAt(0);
+
+  /// The build version number from the `pubspec.yaml` file. Can be null.
+  int get buildNumber {
+    final String value = appVersion?.split('+')?.elementAt(1);
```

Comment:
Fails with `RangeError`, if `appVersion` contains no `+`.

Reply:
jein. It is correct that it fails with a `RangeError` if there is no `+` but in this case, the `_versionPattern` has no match (line 59) and `appVersion` is `null`. So in the end, the `RangeError` can't happen.